component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/util/CoreUtils.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2017 WSO2 Inc. (http://wso2.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.wso2.extension.siddhi.gpl.execution.streamingml.util;
+
+import org.wso2.extension.siddhi.gpl.execution.streamingml.classification.hoeffdingtree.util.AdaptiveHoeffdingTreeModel;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
+import org.wso2.siddhi.query.api.definition.AbstractDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Common utils for Streaming Machine Learning tasks.
+ */
+public class CoreUtils {
+    private static final List<Attribute.Type> numericTypes = Arrays.asList(Attribute.Type.INT,
+            Attribute.Type.DOUBLE, Attribute.Type.LONG, Attribute.Type.FLOAT);
+
+    private static final List<Attribute.Type> labelTypes = Arrays.asList(Attribute.Type.STRING, Attribute.Type.BOOL);
+
+
+    /**
+     * Index of the Maximum from double array
+     *
+     * @param doubles
+     * @return index of the maximum
+     */
+    public static int argMaxIndex(double[] doubles) {
+        double maximum = 0.0D;
+        int maxIndex = 0;
+        for (int i = 0; i < doubles.length; ++i) {
+            if (i == 0 || doubles[i] > maximum) {
+                maxIndex = i;
+                maximum = doubles[i];
+            }
+        }
+        return maxIndex;
+    }
+
+    /**
+     * Maximum from the double array
+     *
+     * @param doubles
+     * @return
+     */
+    public static double argMax(double[] doubles) {
+        double maximum = 0.0D;
+        for (int i = 0; i < doubles.length; ++i) {
+            if (i == 0 || doubles[i] > maximum) {
+                maximum = doubles[i];
+            }
+        }
+        return maximum;
+    }
+
+
+    /**
+     * @param inputDefinition
+     * @param attributeExpressionExecutors
+     * @param startIndex                   starting index
+     * @param noOfFeatures
```

Comment:
Same as above, trying to explain the method if you are to use java doc comments.

Reply:
Fixed in [#d65bc3a9237b1178ce3d2d50ac0fab55709e2e0a](https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/16/commits/d65bc3a9237b1178ce3d2d50ac0fab55709e2e0a)